### ISSUE TITLE
Remove bootstrap from login page

### ISSUE THREAD

**Issue by klopfdreh** [spring-cloud/spring-cloud-dataflow-ui#1887]
As mentioned here https://github.com/spring-cloud/spring-cloud-dataflow-ui/issues/1867 - bootstrap shouldn't be used and removed.

When I tested Spring Cloud Data Flow Server 2.10.0-M2 and used a custom `DataflowOAuthSecurityConfiguration` for CSP settings, I found the login page to still use bootstrap.

![image](https://user-images.githubusercontent.com/980773/192192189-dc8b4f2a-fe6f-4e1d-a620-4a2fd35756bd.png)

**Comment by claudiahub** [comment]
@klopfdreh Thanks Tobias for finding this,
I did some investigation on the Angular repo.
It seems we're not injecting directly from Dataflow UI those two css files,
Do you perhaps have a target URL for the page located the above screenshot ? 
so I can then explore more outside the Angular context

Thanks

**Comment by klopfdreh** [comment]
It is just /login with OAuth2 configured. I am also going to search for it. 👍

**Comment by klopfdreh** [comment]
Yes but the default redirection '{baseUrl}/login/oauth2/code/{registrationId}' points to the scdf server itself. I guess it is the default login page of Spring Security OAuth2 / Spring Security.

**Comment by claudiahub** [comment]
Yep, same guess on my side. So I believe this same issue should be moved to Spring Security.

**Comment by klopfdreh** [comment]
Found it (I guess) - the class is called `LoginPageGeneratingWebFilter` and it includes Bootstrap CSS here:

https://github.com/spring-projects/spring-security/blob/main/web/src/main/java/org/springframework/security/web/server/ui/LoginPageGeneratingWebFilter.java#L98

It is also responsible for OAuth2 authentication, because it provides `setOauth2AuthenticationUrlToClientName`:

https://github.com/spring-projects/spring-security/blob/main/web/src/main/java/org/springframework/security/web/server/ui/LoginPageGeneratingWebFilter.java#L59

and it matches `/login`:

https://github.com/spring-projects/spring-security/blob/main/web/src/main/java/org/springframework/security/web/server/ui/LoginPageGeneratingWebFilter.java#L49

**Comment by klopfdreh** [comment]
So based on the things we found out I would suggest:
1. Configure the backend to show a custom login form as mentioned in my previous commit and by the documentation of Spring Security
2. Create new Rest Endpoints to provide the required authorization data like registrations and so on to the Angular frontend
3. Create a new folder in spring-cloud-data-flow-ui called login and place in there the pages to be shown at login which are using the data from the rest endpoints.

Because it is a login page, it should be excluded from redirection to the login page from the UI. Also be aware of CSRF Tokens which are checked by Spring Security.

the good thing is that you have full control over the appearance of the login and you can design it the SCDF-way.

**Comment by onobc** [comment]
Thanks for the analysis and suggested plan @klopfdreh . I am marking this for team meeting and we will discuss further. I will keep you posted on the direction w/ an update here in this ticket.

**Comment by klopfdreh** [comment]
Thanks again! 😀 I guess it is a mediocre effort but improves the appearance a lot. If I can assist just let me know.

**Comment by oodamien** [comment]
From my understanding, we have to customise Spring Security Settings (create a custom login page).
This should be on the server side.
@corneil We don't need to import all the App CSS. A simple CSS file should be enough.

**Comment by klopfdreh** [comment]
> Investigating options for customising login page while remaining compatible with Spring Security variations without replicating all of Spring Security Filters etc.

Great! Thanks a lot for all the effort. May I don't know all places which are relevant for those changes, but what I think this would be a good place: https://github.com/spring-cloud/spring-cloud-dataflow/blob/main/spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/config/DataflowOAuthSecurityConfiguration.java - in here you can permit all requests to the login page `/login` like:
```java
http.authorizeRequests().antMatchers("/login*").permitAll();
```
Because the context path is also included in this pattern as it only means / with context root - this should work. As I saw here: https://docs.spring.io/spring-cloud-dataflow/docs/current/reference/htmlsingle/#configuration-security-oauth2 - `As of Spring Cloud Data Flow 2.0, OAuth2 is the only mechanism for providing authentication and authorization.`

Further more it would be great when this login screen could be used:

![image](https://user-images.githubusercontent.com/980773/200746176-3b2f3b46-2782-4eba-896a-a1d04868fcc7.png)

I created a prototype:

![image](https://user-images.githubusercontent.com/980773/200746528-9d695aef-957b-4e62-a67d-ebb07018d730.png)

This login page is shown when I press logout in the Spring Cloud Data Flow UI and is already styled in a nice way.

**Comment by klopfdreh** [comment]
@onobc all right! Thanks for the information!
